src/engraving/libmscore/scoreorder.cpp

```diff
@@ -226,6 +291,89 @@ ScoreGroup ScoreOrder::getGroup(const QString family, const QString instrumentGr
     return unsortedScoreGroup;
 }
 
+//---------------------------------------------------------
+//   instrumentSortingIndex
+//---------------------------------------------------------
+
+int ScoreOrder::instrumentSortingIndex(const QString& instrumentId, bool isSoloist) const
+{
+    static const QString SoloistsGroup("<soloists>");
+    static const QString UnsortedGroup("<unsorted>");
+
+    enum class Priority {
+        Undefined,
+        Unsorted,
+        UnsortedGroup,
+        Family,
+        Soloist
+    };
+
+    InstrumentIndex ii { searchTemplateIndexForId(instrumentId) };
+    if (!ii.instrTemplate) {
+        return 0;
+    }
+
+    QString family = instrumentMap.contains(instrumentId) ? instrumentMap[instrumentId].id : ii.instrTemplate->familyId();
+
+    int index = groups.size();
+
+    auto calculateIndex = [instrumentId, &ii](int index) {
+        return index * ii.templateCount + ii.instrTemplate->sequenceOrder;
+    };
+
+    Priority priority = Priority::Undefined;
+
+    for (int i = 0; i < groups.size(); ++i) {
+        const ScoreGroup& sg = groups[i];
+
+        if ((sg.family == SoloistsGroup) && isSoloist) {
+            return calculateIndex(i);
+        } else if ((priority < Priority::Family) && (sg.family == family)) {
+            index = i;
+            priority = Priority::Family;
+        } else if ((priority < Priority::UnsortedGroup) && (sg.family == UnsortedGroup)
+                   && (sg.unsorted == ii.instrTemplate->groupId)) {
+            index = i;
+            priority = Priority::UnsortedGroup;
+        } else if ((priority < Priority::Unsorted) && (sg.family == UnsortedGroup)) {
+            index = i;
+            priority = Priority::Unsorted;
+        }
+    }
+
+    return calculateIndex(index);
+}
+
+//---------------------------------------------------------
+//   isScoreOrder
+//---------------------------------------------------------
+
+bool ScoreOrder::isScoreOrder(const QList<int>& indices) const
+{
+    if (isCustom()) {
+        return true;
+    }
+
+    int prvIndex { -1 };
+    for (int curIndex : indices) {
+        if (curIndex < prvIndex) {
+            return false;
+        }
+        prvIndex = curIndex;
+    }
+    return true;
+}
+
+bool ScoreOrder::isScoreOrder(const Score* score) const
+{
+    QList<int> indices;
+    for (Part* part : score->parts()) {
```

Comment:
Please, add const for this pointer:

for (const Part* part : score->parts())

Reply:
Done